### ISSUE TITLE
Linkage errors with version 10.1.6 and Ceylon SDK modules

### ISSUE THREAD

**Issue by fwgreen** [ebean-orm/ebean-agent#48]
If I use any Ceylon SDK modules in a project with Ebean Agent 10.1.6, I get linkage errors:
```
Exception in thread "main" java.lang.LinkageError: loader (instance of  com/redhat/ceylon/launcher/CeylonClassLoader): attempted  duplicate class definition for name: "ceylon/language/Float"
```
Version 10.1.2 works without complaints, and 10.1.5 also works, but has class not found errors:
```
getCommonSuperClass java/lang/Object :ceylon/language/String
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.String
getCommonSuperClass ceylon/language/Float :ceylon/language/ParseException
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.Float
getCommonSuperClass ceylon/language/equal_ :ceylon/language/larger_
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.equal_
getCommonSuperClass ceylon/language/smaller_ :java/lang/Object
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.smaller_
getCommonSuperClass java/lang/Object :ceylon/language/String
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.String
getCommonSuperClass ceylon/collection/LinkedCell :ceylon/collection/CachingCell
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.collection.LinkedCell
getCommonSuperClass ceylon/collection/LinkedCell :java/lang/Object
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.collection.LinkedCell
getCommonSuperClass ceylon/collection/LinkedCellIterator :ceylon/collection/CachingStoreIterator
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.collection.LinkedCellIterator
getCommonSuperClass ceylon/language/empty_ :ceylon/collection/HashMap
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.empty_
getCommonSuperClass java/lang/Object :ceylon/collection/HashMap
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.collection.HashMap
getCommonSuperClass java/lang/Throwable :java/lang/invoke/MethodHandles$Lookup
getCommonSuperClass ceylon/language/List :ceylon/collection/ArrayList
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.List
getCommonSuperClass ceylon/collection/ArrayList :ceylon/language/List
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.collection.ArrayList
getCommonSuperClass ceylon/json/Object :ceylon/language/Sequential
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.json.Object
getCommonSuperClass java/lang/Object :ceylon/language/Sequential
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.Sequential
getCommonSuperClass ceylon/language/empty_ :ceylon/language/ArraySequence
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.empty_
getCommonSuperClass ceylon/language/ArraySequence :ceylon/language/empty_
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.ArraySequence
getCommonSuperClass ceylon/language/Sequence$impl$Repeat :ceylon/language/empty_
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.Sequence$impl$Repeat
getCommonSuperClass ceylon/language/JoinedSequence :ceylon/language/Sequence
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.JoinedSequence
getCommonSuperClass ceylon/language/Sequence :ceylon/language/Empty
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.Sequence
getCommonSuperClass ceylon/language/Sequential :ceylon/language/empty_
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.Sequential
getCommonSuperClass ceylon/language/ArraySequence :ceylon/language/empty_
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.ArraySequence
getCommonSuperClass ceylon/language/Sequence :ceylon/language/ArraySequence
getCommonSuperClass err: java.lang.ClassNotFoundException: ceylon.language.Sequence
```
Being as 10.1.2 works fine for me, this is more or less a tracking issue as it is probably a problem between ASM and Ceylon, but I lack the expertise to figure it out.

**Comment by rbygrave** [comment]
Can you retry with 10.3.1 ?  Thanks.

**Comment by fwgreen** [comment]
Sorry for the late response.  I still have the error with 10.3.1.

**Comment by fwgreen** [comment]
This problem has been resolved as version 11.11.1.  The constant fine tuning of this library has paid unexpected dividends.  Thanks! :smile:
